networkx/algorithms/similarity.py

```diff
@@ -1167,13 +1159,11 @@ def get_edit_paths(
     # debug_print(Ce.C)
     # debug_print()
 
+    @dataclass
     class MaxCost:
-        def __init__(self):
-            # initial upper-bound estimate
-            # NOTE: should work for empty graph
-            self.value = Cv.C.sum() + Ce.C.sum() + 1
+        value: ...
 
-    maxcost = MaxCost()
+    maxcost = MaxCost(Cv.C.sum() + Ce.C.sum() + 1)
```

Comment:
Does this even need to be a data class? Since it's a single value, can't we just use a variable here?

Reply:
I also wonder if MaxCost needs to be used.  it seems to be a one variable dataclass.

Reply:
There is a way to remove it, but then we need to put in `nonlocal` for the maxcost val. Python doesn't pass around variables but apparently objects are fine? Updated the code in https://github.com/networkx/networkx/pull/5831/commits/67db6c0c1129ea6dcc90c7eb72b75555c178c2c7


More fun python things:
``` python
In [1]: from dataclasses import dataclass

In [2]: @dataclass
   ...: class Test:
   ...:     val: int
   ...:

In [3]: def funky_func():
   ...:     x_val = x_val + 1
   ...:     print(x_val)
   ...:     return x_val
   ...:

In [4]: def funky_func_class():
   ...:     x.val = x.val + 1
   ...:     print(x.val)
   ...:     return x.val
   ...:

In [5]: x = Test(1)

In [6]: funky_func_class()
2
Out[6]: 2

In [7]: x = 1

In [8]: funky_func()
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
Input In [9], in <cell line: 1>()
----> 1 funky_func()

Input In [3], in funky_func()
      1 def funky_func():
----> 2     x_val = x_val + 1
      3     print(x_val)
      4     return x_val

UnboundLocalError: local variable 'x_val' referenced before assignment
```